tensorflow/core/kernels/mkl/mkl_matmul_op_fused.cc

```diff
@@ -322,34 +342,630 @@ class MklFusedMatMulOp : public MklDnnMatMulOpBase<T, void, T> {
     }
   }
 
- private:
+ protected:
+  virtual void GetScaledBias(
+      OpKernelContext*,
+      std::shared_ptr<dnnl::inner_product_forward::primitive_desc>&,
+      const Tensor&, Tensor*, void**) {}
+
   bool fuse_add_ = false;
   bool transpose_a_;
   bool transpose_b_;
-  float leakyrelu_alpha = 0.2;
+  float leakyrelu_alpha_ = 0.2;
   std::vector<string> fused_ops_;
-  const int kInputIndex_Add = 3;
+  int input_idx_add_ = 3;
   const int kOutputIndex_Dst = 0;
-};  // namespace tensorflow
+#ifdef DNNL_AARCH64_USE_ACL
+  const int kWeightTensorHashLength = 1024;
+#endif
+};
+
+namespace {
+
+enum class FusedComputationType {
+  kUndefined,
+  kBiasAdd,
+  kBiasAdd_Dequantize,
+  kBiasAdd_Requantize,
+  kBiasAdd_Activation,
+  kBiasAdd_Activation_Dequantize,
+  kBiasAdd_Activation_Requantize,
+  kBiasAdd_Add,
+  kBiasAdd_Add_Dequantize,
+  kBiasAdd_Add_Requantize,
+};
+
+struct FusedComputationPattern {
+  FusedComputationType fused_computation;
+  std::vector<string> fused_ops;
+};
+
+}  // namespace
+
+// OneDNN uses post-ops to implement different kind of fusions. The category of
+// each individual post-op can be inferred from the fused_ops attribute. The
+// following enum is used to identify list of required post-ops.
+enum class PostOpKind { kActivation, kSum, kOutputScale, kLinear };
+
+template <typename Device, typename T1, typename T2, typename Tbias,
+          typename Toutput, typename U, bool native_format = true>
+class QuantizedFusedMatMulOp
+    : public MklFusedMatMulOp<Device, T1, T2, Tbias, Toutput, U,
+                              native_format> {
+ protected:
+  string input_quant_mode_;   // 0-th input
+  string output_quant_mode_;  // 0-th output
+  string activation_type_;    // Activation op type
+
+  // Initialize minmax tensor indices with default values for the most common
+  // cases.
+  int input_min_idx_ = 3;
+  int input_max_idx_ = 4;
+  int weight_min_idx_ = 5;
+  int weight_max_idx_ = 6;
+
+  struct PostOpInfo {
+    PostOpKind post_op_kind;
+    struct OperandInfo {
+      int idx = -1;  // Operand tensor index if needed by a post-op.
+      // Indices of min and max value tensors, if the operand is quantized.
+      gtl::InlinedVector<int, 4> min_max_indices;
```

Comment:
These should all be replaced by `absl::InlinedVector` now.

Reply:
Fixed.